src/placement/pl_map_common.c

```diff
@@ -196,23 +197,28 @@ int
 remap_list_fill(struct pl_map *map, struct daos_obj_md *md,
 		struct daos_obj_shard_md *shard_md, uint32_t r_ver,
 		uint32_t *tgt_id, uint32_t *shard_idx,
-		unsigned int array_size, int myrank, int *idx,
+		unsigned int array_size, int *idx,
 		struct pl_obj_layout *layout, d_list_t *remap_list,
 		bool fill_addition)
 {
 	struct failed_shard     *f_shard;
 	struct pl_obj_shard     *l_shard;
-	int                     rc = 0;
 
 	d_list_for_each_entry(f_shard, remap_list, fs_list) {
 		l_shard = &layout->ol_shards[f_shard->fs_shard_idx];
 
 		if (f_shard->fs_fseq > r_ver)
 			break;
 
+		/**
+		 * NB: due to colrelocation, the data on healthy
+		 * shard might need move to, so let's include
+		 * UPIN status here as well.
+		 */
 		if (f_shard->fs_status == PO_COMP_ST_DOWN ||
 		    f_shard->fs_status == PO_COMP_ST_UP ||
 		    f_shard->fs_status == PO_COMP_ST_DRAIN ||
+		    f_shard->fs_status == PO_COMP_ST_UPIN ||
```

Comment:
This is only for reintegration, right? Will it cause problem for rebuild or addtion case?

Reply:
Hmm, if layout algothrim is correct, then it is impossible to get here with UPIN node, I think. Addition is possible I think. 

But anyway, we should move healthy data shard in any case if the layout is being changed.

Reply:
If that's the case, I think all the checks are not necessary, we should just fill the target id array for every item in the remap_list. (Probably with some assertion on the target state)

I'm fine with cleanup it later.